examples/hello_world/CMakeLists.txt

```diff
@@ -10,26 +10,29 @@ find_package(Threads REQUIRED)
 
 set(VSOMEIP_NAME "vsomeip3")
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14")
+# create_target("executable")
+function(create_target executable)
+    add_library(vsomeip_hello_world_${executable} INTERFACE)
+    target_sources(vsomeip_hello_world_${executable} INTERFACE
```

Comment:
Using `target_sources` is awesome, make sure you change the `cmake_minimum_required` in line 6 to at least CMake 3.1 (I'd say use at least 3.15)